### DISCUSSION TITLE
loader till metamask tranaction finalized after confirming transaction on wallet using wagmi hooks

### DISCUSSION THREAD

**Discussion by abhi8960git** [wevm/wagmi#4167] [General]
i tried to use wagmi hooks such that when i click on a write fn using wagmi hooks it will popup metamask and i will confirm transaction , but i want to add loader in ui till the transaction fully finalized , but i am not able to do that using wagmi hooks ??

is it possible with wagmi hooks

**Comment by tmm** [comment]
Not sure what you code looks like, but you should be able to use [`writeContract`](https://wagmi.sh/react/api/hooks/useWriteContract).

```tsx
import { useWriteContract } from 'wagmi'

const { isPending, writeContract } = useWriteContract()

<button
  disabled={isPending}
  onClick={() => writeContract(...)}
>
  {isPending ? 'Write' : 'Check Wallet'}
</button>
```

**Reply by abhi8960git** [reply]
lets say this is code to mint NFT i clicked on mint button and it will popup metamask and from there i will confirm transaction , but still transaction is pending it yet not get confirmed or failed , i want that is is possible i can use wagmi a variable that will become true only when after confirming in metamask a transaction when a transaction is finally confirmed or failed then it will become true so that i can use that variable as loader for ui.

**Reply by tmm** [reply]
[`useWaitForTransactionReceipt`](https://wagmi.sh/react/api/hooks/useWaitForTransactionReceipt#usewaitfortransactionreceipt)
